base_ubl/models/ubl.py

```diff
@@ -0,0 +1,695 @@
+            'email': email_xpath and email_xpath[0].text or False,
+            'website': website_xpath and website_xpath[0].text or False,
+            'phone': phone_xpath and phone_xpath[0].text or False,
+            }
+        address_xpath = party_node.xpath('cac:PostalAddress', namespaces=ns)
+        if address_xpath:
+            address_dict = self.ubl_parse_address(address_xpath[0], ns)
+            partner_dict.update(address_dict)
+        return partner_dict
+
+    @api.model
+    def ubl_parse_address(self, address_node, ns):
+        country_code_xpath = address_node.xpath(
+            'cac:Country/cbc:IdentificationCode',
+            namespaces=ns)
+        country_code = country_code_xpath and country_code_xpath[0].text\
+            or False
+        state_code_xpath = address_node.xpath(
+            'cbc:CountrySubentityCode', namespaces=ns)
+        state_code = state_code_xpath and state_code_xpath[0].text or False
+        zip_xpath = address_node.xpath('cbc:PostalZone', namespaces=ns)
+        zip = zip_xpath and zip_xpath[0].text and\
+            zip_xpath[0].text.replace(' ', '') or False
+        address_dict = {
+            'zip': zip,
+            'state_code': state_code,
+            'country_code': country_code,
+            }
+        return address_dict
+
+    @api.model
+    def ubl_parse_delivery(self, delivery_node, ns):
+        party_xpath = delivery_node.xpath('cac:DeliveryParty', namespaces=ns)
+        if party_xpath:
+            partner_dict = self.ubl_parse_party(party_xpath[0], ns)
+        else:
+            partner_dict = {}
+        delivery_address_xpath = delivery_node.xpath(
+            'cac:DeliveryLocation/cac:Address', namespaces=ns)
+        if not delivery_address_xpath:
+            delivery_address_xpath = delivery_node.xpath(
+                'cac:DeliveryAddress', namespaces=ns)
+        if delivery_address_xpath:
+            address_dict = self.ubl_parse_address(
+                delivery_address_xpath[0], ns)
+        else:
+            address_dict = {}
+        delivery_dict = {
+            'partner': partner_dict,
+            'address': address_dict,
+            }
+        return delivery_dict
+
+    def ubl_parse_incoterm(self, delivery_term_node, ns):
+        incoterm_xpath = delivery_term_node.xpath("cbc:ID", namespaces=ns)
+        if incoterm_xpath:
+            incoterm_dict = {'code': incoterm_xpath[0].text}
+            return incoterm_dict
+        return {}
+
+    def ubl_parse_product(self, line_node, ns):
+        barcode_xpath = line_node.xpath(
+            "cac:Item/cac:StandardItemIdentification/cbc:ID[@schemeID='GTIN']",
+            namespaces=ns)
+        code_xpath = line_node.xpath(
+            "cac:Item/cac:SellersItemIdentification/cbc:ID", namespaces=ns)
+        product_dict = {
+            'barcode': barcode_xpath and barcode_xpath[0].text or False,
+            'code': code_xpath and code_xpath[0].text or False,
```

Comment:
```suggestion
            'default_code': code_xpath and code_xpath[0].text or False,
```

Reply:
should this change be backported to v11?